client/web/src/enterprise/batches/detail/changesets/BatchChangeChangesets.tsx

```diff
@@ -63,7 +68,42 @@ export const BatchChangeChangesets: React.FunctionComponent<Props> = ({
     queryExternalChangesetWithFileDiffs,
     expandByDefault,
     onlyArchived,
+    enableSelect,
 }) => {
+    const [selectedChangesets, setSelectedChangesets] = useState<Set<string>>(new Set())
+    const onSelect = (id: string, selected: boolean): void => {
+        if (selected) {
+            setSelectedChangesets(() => new Set([...selectedChangesets, id]))
+        } else {
+            setSelectedChangesets(new Set([...selectedChangesets].filter(selectedId => selectedId !== id)))
+        }
+    }
+
+    const deselectAll = useCallback((): void => setSelectedChangesets(new Set([])), [setSelectedChangesets])
+    const changesetSelected = (id: string): boolean => selectedChangesets.has(id)
+
+    const [isSubmittingSelected, setIsSubmittingSelected] = useState<boolean | Error>(false)
+    const onSubmitSelected = useCallback(async () => {
+        if (
+            !confirm(
+                `Are you sure you want to detach ${selectedChangesets.size} ${pluralize(
+                    'changesets',
+                    selectedChangesets.size
+                )}?`
+            )
+        ) {
+            return
+        }
+        setIsSubmittingSelected(true)
+        try {
+            await detachChangesets(batchChangeID, [...selectedChangesets])
+            deselectAll()
+            telemetryService.logViewEvent('BatchChangeDetailsPageDetachArchivedChangesets')
+        } catch (error) {
+            setIsSubmittingSelected(error)
```

Comment:
```suggestion
            setIsSubmittingSelected(asError(error))
```

ensures this is actually an error and recognized correctly by `isErrorLike`